nms/app/views/domain-proxy/CbsdEdit.js

```diff
@@ -0,0 +1,425 @@
+
+import Text from '../../theme/design-system/Text';
+import CbsdContext from '../../components/context/CbsdContext';
+import type { cbsd, mutable_cbsd } from '../../../generated/MagmaAPIBindings';
+import { AltFormField, AltFormFieldSubheading } from '../../components/FormField';
+import DialogTitle from '../../theme/design-system/DialogTitle';
+import { colors, typography } from '../../theme/default';
+
+const useStyles = makeStyles(theme => ({
+  appBarBtn: {
+    color: colors.primary.white,
+    background: colors.primary.comet,
+    fontFamily: typography.button.fontFamily,
+    fontWeight: typography.button.fontWeight,
+    fontSize: typography.button.fontSize,
+    lineHeight: typography.button.lineHeight,
+    letterSpacing: typography.button.letterSpacing,
+
+    '&:hover': {
+      background: colors.primary.mirage,
+    },
+  },
+  tabBar: {
+    backgroundColor: colors.primary.brightGray,
+    color: colors.primary.white,
+  },
+  frequencyPreferencesContent: {
+    marginTop: theme.spacing(1),
+  },
+  bandwidthTitleContainer: {
+    height: theme.spacing(4),
+  },
+  frequencyPreferencesIcon: {
+    padding: 0,
+  },
+  addFrequencyIconContainer: {
+    textAlign: 'right',
+  },
+}));
+
+type ButtonProps = {
+  title: string,
+  isLink: boolean,
+};
+
+export function AddEditCbsdButton(props: ButtonProps) {
+  const classes = useStyles();
+  const [open, setOpen] = useState(false);
+
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  return (
+    <>
+      <CbsdAddEditDialog open={open} onClose={handleClose} />
+      {props.isLink ? (
+        <Button
+          component="button"
+          variant="text"
+          onClick={handleOpen}>
+          {props.title}
+        </Button>
+      ) : (
+        <Button
+          variant="text"
+          className={classes.appBarBtn}
+          onClick={handleOpen}>
+          {props.title}
+        </Button>
+      )}
+    </>
+  );
+}
+
+type DialogProps = {
+  open: boolean,
+  onClose: () => void,
+  cbsd?: cbsd,
+};
+
+type BandwidthEnum = 5 | 10 | 15 | 20;
+
+type CbsdFormData = {
+  serialNumber: string,
+  fccId: string,
+  userId: string,
+  minPower: number,
+  maxPower: number,
+  numberOfAntennas: number,
+  antennaGain: number,
+  bandwidthMhz: BandwidthEnum,
+  frequenciesMhz: number[],
+};
+
+const convertToCbsdFormData = (cbsd?: cbsd): CbsdFormData => {
+  return {
+    serialNumber: cbsd?.serial_number || '',
+    fccId: cbsd?.fcc_id || '',
+    userId: cbsd?.user_id || '',
+    minPower: cbsd?.capabilities?.min_power || 0,
+    maxPower: cbsd?.capabilities?.max_power || 0,
+    numberOfAntennas: cbsd?.capabilities?.number_of_antennas || 0,
+    antennaGain: cbsd?.capabilities?.antenna_gain || 0,
+    bandwidthMhz: cbsd?.frequency_preferences?.bandwidth_mhz || 5,
+    frequenciesMhz: cbsd?.frequency_preferences?.frequencies_mhz || [0],
+  }
+}
+
+export function CbsdAddEditDialog(props: DialogProps) {
```

Comment:
We should have some place holders here. Ideally with example input
![Screenshot 2022-05-05 at 18 24 56](https://user-images.githubusercontent.com/102796295/166969291-ed1b692b-3256-49dc-b47f-f1e99c395243.png)
.

Reply:
Fixed

Reply:
I still see a bunch of blank fields. Checkout the "Add Gateway" Dialog for how this should ideally look like. 
It is especially useful if we require a certain format, but I can't judge if this is the case here.
![Screenshot 2022-06-13 at 09 12 35](https://user-images.githubusercontent.com/102796295/173300006-98585731-b310-4dd5-a44d-563870a492d0.png)

Reply:
Yes. Although the zero for Number of Antennas doesn't have the wrong color imho. If if is a valid pre-filled value it is fine. If it isn't valid it should be replaced with something like `Enter number of antennas`